src/shared/mkfs-util.c

```diff
@@ -444,32 +453,40 @@ int make_filesystem(
                 }
 
                 /* mkfs.vfat does not have a --quiet option so let's redirect stdout to /dev/null instead. */
-                stdio_fds[1] = -EBADF;
+                if (quiet)
+                        stdio_fds[1] = -EBADF;
 
-        } else if (streq(fstype, "swap"))
-                /* TODO: add --quiet here if
-                 * https://github.com/util-linux/util-linux/issues/1499 resolved. */
+        } else if (streq(fstype, "swap")) {
+                /* TODO: add --quiet once util-linux v2.38 is available everywhere. */
 
                 argv = strv_new(mkfs,
                                 "-L", label,
                                 "-U", vol_id,
                                 node);
 
-        else if (streq(fstype, "squashfs")) {
+                if (quiet)
+                        stdio_fds[1] = -EBADF;
+
+        } else if (streq(fstype, "squashfs")) {
 
                 argv = strv_new(mkfs,
                                 root, node,
                                 "-noappend");
 
                 /* mksquashfs -quiet option is pretty new so let's redirect stdout to /dev/null instead. */
-                stdio_fds[1] = -EBADF;
+                if (quiet)
+                        stdio_fds[1] = -EBADF;
 
-        } else if (streq(fstype, "erofs"))
+        } else if (streq(fstype, "erofs")) {
 
                 argv = strv_new(mkfs,
                                 "-U", vol_id,
                                 node, root);
-        else
+
+                if (quiet && strv_extend(&argv, "--quiet") < 0)
```

Comment:
this is new, was that intended?

Reply:
Yes it is, the commit mentioned that we add support for --quiet for some filesystems that didn't have it until now.